app.py

```diff
@@ -53,11 +55,40 @@ def inject_configurations():
 # Routes
 @app.route('/')
 @login_required
-def index():  # put application's code here
+def index():
     current_year = get_current_year()
+    year = db.session.query(Year).filter_by(year=current_year).first()
     user = db.session.query(User).filter_by(email=session['email']).first()
-    courses_teacher = db.session.query(Course).filter_by(year=current_year).join(Teacher).filter(Teacher.user_id == user.id).all()
-    return render_template("home.html", user=user, courses=courses_teacher, evaluations=user.evaluations)
+    researcher = db.session.query(Researcher).filter_by(user_id=user.id).first()
+    teacher = db.session.query(Teacher).filter_by(user_id=user.id).first() if user.is_teacher else None
```

Comment:
I feel like you could use conditionals on the user role to avoid unnecessary fetching from the DB and computing.

ex : 
```py
if user.is_teacher : 
    # fetch and computing related to teacher
elif user.is_researcher : 
    # fetch and computing related to researcher
elif ... : 
   # ...
```